lib/modules/datasource/go/releases-goproxy.ts

```diff
@@ -72,7 +72,9 @@ export class GoProxyDatasource extends Datasource {
           } catch (err) {
             logger.trace({ err }, `Can't get datasource for ${packageName}`);
           }
-          break;
+          if (result.releases.length) {
+            break;
+          }
```

Comment:
Seems like this change must've been done irrespective of `res.tags` presense or absense?

But now, when we check `res.tags`, non-empty `res.releases` values should take precedence over just non-empty `res.tags` values, right?

Reply:
Yep!  I wanted to retain the original logic that results with `releases` should **always** take priority over results without them.  (That's because when `releases` is empty, `res.tags.latest` is only a digest, but we'd rather return tagged releases if any proxy knows about them.)

I've also [added some more background context to the original discussion thread](https://github.com/renovatebot/renovate/discussions/24577#discussioncomment-7083096) that might help.

Reply:
I've pushed one more change (3d28a97) to make the intention more clear.